tests/org.eclipse.passage.lic.net.tests/src/org/eclipse/passage/lic/net/tests/io/SafePayloadTest.java

```diff
@@ -16,33 +16,67 @@
 import static org.junit.Assert.assertTrue;
 import static org.junit.Assert.fail;
 
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
 import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
 
 import org.eclipse.passage.lic.api.LicensingException;
 import org.eclipse.passage.lic.api.io.KeyKeeper;
 import org.eclipse.passage.lic.base.io.FileKeyKeeper;
 import org.eclipse.passage.lic.base.io.MD5Hashes;
 import org.eclipse.passage.lic.internal.net.io.SafePayload;
+import org.junit.ClassRule;
 import org.junit.Test;
+import org.junit.jupiter.api.function.ThrowingSupplier;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameter;
+import org.junit.runners.Parameterized.Parameters;
 
+@RunWith(Parameterized.class)
 public final class SafePayloadTest {
+	@ClassRule
+	public static TemporaryFolder tempFolder = new TemporaryFolder();
+
+	@Parameter
+	public ThrowingSupplier<KeyKeeper> keyKeeper2;
+
+	@Parameters
+	public static List<ThrowingSupplier<KeyKeeper>> keyKeepers() {
+		return Arrays.asList(SafePayloadTest::keeper, SafePayloadTest::keeperWithDifferentLineDelimiter);
+	}
 
 	@Test
-	public void symmetric() {
+	public void symmetric() throws Throwable {
 		String original = "S0me sophisticäted Str!ng"; //$NON-NLS-1$
 		try {
-			byte[] encoded = new SafePayload(keerper(), new MD5Hashes()).encode(original.getBytes());
+			byte[] encoded = new SafePayload(keeper(), new MD5Hashes()).encode(original.getBytes());
 			assertTrue(encoded.length > 0);
 			System.out.println(new String(encoded));
-			byte[] decoded = new SafePayload(keerper(), new MD5Hashes()).decode(encoded);
+			byte[] decoded = new SafePayload(keyKeeper2.get(), new MD5Hashes()).decode(encoded);
 			assertTrue(decoded.length > 0);
 			assertEquals(original, new String(decoded));
 		} catch (LicensingException e) {
 			fail("Not intended to fail on valid data"); //$NON-NLS-1$
 		}
 	}
 
-	private KeyKeeper keerper() throws LicensingException {
+	private static KeyKeeper keeper() throws LicensingException {
```

Comment:
Vaoid `static` methods by any menas, if it is possible. 
Create couple of package-local classes instead.

Reply:
Done.